OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/ChatRoomWorkshopServiceWithDBTests.cs

```diff
@@ -0,0 +1,321 @@
+                {
+                    Id = item.Workshop.Id,
+                    Title = item.Workshop.Title,
+                    ProviderId = item.Workshop.ProviderId,
+                    ProviderTitle = item.Workshop.ProviderTitle,
+                },
+                ParentId = item.ParentId,
+                Parent = new ParentInfoForChatList()
+                {
+                    Id = item.Parent.Id,
+                    UserId = item.Parent.UserId,
+                    FirstName = item.Parent.User.FirstName,
+                    MiddleName = item.Parent.User.MiddleName,
+                    LastName = item.Parent.User.LastName,
+                    Gender = item.Parent.Gender,
+                    Email = item.Parent.User.Email,
+                    PhoneNumber = item.Parent.User.PhoneNumber,
+                },
+            });
+
+        var expectedResultEntities = expectedRooms
+            .Select(r => mapper.Map<ChatRoomWorkshopDtoWithLastMessage>(r))
+            .ToList();
+
+        // Act
+        var result = await roomService
+            .GetChatRoomByFilter(new ChatWorkshopFilter(), existingProviderId, searchMessagesForProvider);
+
+        // Assert
+        Assert.That(result is not null);
+        Assert.AreEqual(expectedResultEntities.Count, result.TotalAmount);
+        expectedResultEntities.Should().BeEquivalentTo(result.Entities);
+    }
+
+    [Test]
+    public async Task GetChatRoomByFilter_WhenUserIsValidParent_ShouldReturnRightSearchResult()
+    {
+        // Arrange
+        var existingParentId = parents[0].Id;
+        var searchMessagesForProvider = false;
+        var expectedRooms = rooms
+            .Where(item => item.ParentId == existingParentId)
+            .Select(item => new ChatRoomWorkshopForChatList()
+            {
+                Id = item.Id,
+                WorkshopId = item.WorkshopId,
+                Workshop = new WorkshopInfoForChatList()
+                {
+                    Id = item.Workshop.Id,
+                    Title = item.Workshop.Title,
+                    ProviderId = item.Workshop.ProviderId,
+                    ProviderTitle = item.Workshop.ProviderTitle,
+                },
+                ParentId = item.ParentId,
+                Parent = new ParentInfoForChatList()
+                {
+                    Id = item.Parent.Id,
+                    UserId = item.Parent.UserId,
+                    FirstName = item.Parent.User.FirstName,
+                    MiddleName = item.Parent.User.MiddleName,
+                    LastName = item.Parent.User.LastName,
+                    Gender = item.Parent.Gender,
+                    Email = item.Parent.User.Email,
+                    PhoneNumber = item.Parent.User.PhoneNumber,
+                },
+            });
+
+        var expectedResultEntities = expectedRooms
+            .Select(r => mapper.Map<ChatRoomWorkshopDtoWithLastMessage>(r))
+            .ToList();
```

Comment:
(Not a call to action right now, just asking for you to reconsider test writing approach) 
You are doubling the method logic in your test. It makes test very fragile to any changes in less relaetd code than test suppose to check - i.e. changes in mapper. 

As for me, for this specific test we could assert that `GetChatRoomByFilter` returns non-empty result (as we have prepared data for this) and that all returned rooms has expected parents id. 

Ofc, other tests could be rewritten as well.

Again, I'm not insisting on any changes right now, just think about it, please

Reply:
Thanks, that's really better. I will change it as you suggested